crates/types/src/cluster_controller.rs

```diff
@@ -34,10 +34,10 @@ pub enum ReplicationStrategy {
 #[serde_as]
 #[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
 pub struct SchedulingPlan {
-    version: Version,
+    pub version: Version,
     // flexbuffers only supports string-keyed maps :-( --> so we store it as vector of kv pairs
     #[serde_as(as = "serde_with::Seq<(_, _)>")]
-    partitions: BTreeMap<PartitionId, TargetPartitionState>,
+    pub partitions: BTreeMap<PartitionId, TargetPartitionState>,
```

Comment:
Why do you need to make the fields public?

Reply:
`sync_routing_information` iterates over the items in `partitions` directly - seemed like I might as well make `version` private at the same time. On reflection, it seems like a more Rust-y approach would be to either implement `IntoIterator` or expose an `iter()` method from `SchedulingPlan` - which it turns out, you already thought about :-) Totally missed the existing `iter()`, will revert!